src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonMetadataServicesConverter.cs

```diff
@@ -61,6 +61,7 @@ internal override bool OnTryWrite(Utf8JsonWriter writer, T value, JsonSerializer
             if (!state.SupportContinuation &&
                 jsonTypeInfo is JsonTypeInfo<T> info &&
                 info.SerializeHandler != null &&
+                !state.CurrentContainsMetadata && // Do not use the fast path if state needs to write metadata.
```

Comment:
I find this a bit confusing. If SerializeHandler is defined we should always come here. SourceGen should not assign SerializeHandler for polimorphic types

Reply:
I'm ok with fixing this post preview since currently SerializeHandler is only assigned by source gen but generally speaking this won't work correctly once users can specify their own handler

Reply:
In this context `CurrentContainsMetadata` will be true if we're calling into a derived converter from a base converter that has already emitted metadata (e.g. `$id` or `$type`). So calling into the fast path is never appropriate.